src/main/java/com/microsoft/azure/vmagent/AzureVMManagementServiceDelegate.java

```diff
@@ -1114,10 +1146,23 @@ public static void terminateVirtualMachine(
                 for (URI diskUri : diskUrisToRemove) {
                     AzureVMManagementServiceDelegate.removeStorageBlob(azureClient, diskUri, resourceGroupName);
                 }
+                for (String id : diskIdToRemove) {
+                    LOGGER.log(Level.INFO, "AzureVMManagementServiceDelegate: terminateVirtualMachine: Removing managed disk with id: {0}", id);
+                    azureClient.disks().deleteById(id);
```

Comment:
cleanLeakedResources doesn't call terminateVirtualMachine. You need to update that method as well to remove the managed disks